### PR TITLE
Lab 3 - Letter frequency

### PR DESCRIPTION
**Pull request by mohankrishna**
Fixes #63 

#### Changes proposed in this PR:

- Implement a function to return mapping of each letter to its frequency
```go
func letters(s string) map[rune]int
```
- Implement a function that returns a sorted slice of strings with elements "{key}:{val}"
```go
func print(m map[rune]int) []string
```
- Add file 03_letters/mohankrishna/main.go which contains implementation of the above mentioned functions
- Add file 03_letters/mohankrishna/main_test.go contins unit tests for the above imlepemntation

### CODE REVIEW

File: 03_letters/mohankrishna/main.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, strings.Join(sortLetters(letters("aba")), "\n"))
+}
+
+func letters(s string) map[rune]int {
+	letterMap := make(map[rune]int)
+	for _, char := range s {
+		letterMap[char]++
```
Comment (juliaogris) [inline 03_letters/mohankrishna/main.go:20]: 👍

**Review by juliaogris — DISMISSED** [review]
LGTM!

File: 03_letters/mohankrishna/main_test.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	sortedStrings []string
+}{
+	{"", map[rune]int{}, []string{}},
+	{"abdsd3$asda$asasdd$sadas",
+		map[rune]int{'a': 7, 'b': 1, 'd': 6, 's': 6, '3': 1, '$': 3},
+		[]string{"$:3", "3:1", "a:7", "b:1", "d:6", "s:6"}},
+	{"!&^#*&+@#*&@+#$#&!*$!$ !^+ &$!*&$^ +$&^!#$ &!$!^$&!# $",
+		map[rune]int{'!': 9, '&': 9, '^': 5, '#': 6, '*': 4, '+': 4, '@': 2, '$': 10, ' ': 5},
+		[]string{" :5", "!:9", "#:6", "$:10", "&:9", "*:4", "+:4", "@:2", "^:5"}},
+	{"🦋😱😈😎🦖🤠😎😈🦖😎🐉🤠🦖🦋🦋🦋😈😎🐉🦖🤠😈🐉😎🤠😈🐉🤠😱🦋",
```
Comment (anzboi) [inline 03_letters/mohankrishna/main_test.go:22]: This works? love it.

**Review by anzboi — DISMISSED** [review]
LGTM!

File: 03_letters/mohankrishna/main.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+func sortLetters(m map[rune]int) []string {
+	keys := []int{}
+	for k := range m {
+		keys = append(keys, int(k))
+	}
+	sort.Ints(keys)
+
+	var sortedKeys = make([]string, len(keys))
```
Comment (camscale) [inline 03_letters/mohankrishna/main.go]: :+1: for pre-allocating the size

you can remove the `var` and use `:=` here, which follows the same style as the other declarations in this function.

File: 03_letters/mohankrishna/main_test.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var tests = []struct {
```
Comment (camscale) [inline 03_letters/mohankrishna/main_test.go:10]: :+1: for table-driven tests. very idiomatic go.

File: 03_letters/mohankrishna/main.go (hunk 3 above)
Comment (mohankrishna) [reply 03_letters/mohankrishna/main.go]: Fixed it!

**Review by camscale — APPROVED** [review]
LGTM

**Pull request merged**